File: frontend/classes/AdShotRunner/Utilities/FileStorageClient.php
<?PHP
/**
* Contains the class for storing and deleting files
*
* @package Adshotrunner
* @subpackage Classes
*/

namespace AdShotRunner\Utilities;

/**
* The FileStorageClient saves and deletes files for system wide (and optionally public) access
*/
class FileStorageClient {

	//---------------------------------------------------------------------------------------
	//---------------------------------- Constants ------------------------------------------
	//---------------------------------------------------------------------------------------	
	const TAGIMAGESCONTAINER = "asr-tagimages";

	//--------------------------------------------------------------------------------------
	//---------------------------------- Static Methods ------------------------------------
	//--------------------------------------------------------------------------------------
	//***************************** Public Static Methods **********************************
	/**
	 * Saves a file in the chosen container with the supplied filename.
	 * 
	 * @param string 	containerName		Name of the container to save the file in
	 * @param string 	fileToSave			Name and optional path of file to save
	 * @param string 	newFilename			Name that the file to save should be stored under
	 */
	static public function saveFile($containerName, $fileToSave, $newFilename) {

		//Verify all arguments were not empty
		if (!$containerName || !$fileToSave || !$newFilename) {return FALSE;}

		//Create the S3 handler object
		$awsFactory = getAWSFactory();
		$s3Handler = $awsFactory->get('s3');

		//Store the file
		$s3Handler->putObject(array( 'Bucket'       => $containerName,
									 'SourceFile'   => $fileToSave,
									 'Key'          => $newFilename));
	}

	/**
	 * Deletes a file from the chosen container
	 * 
	 * @param string 	containerName		Name of the container the file resides in
	 * @param string 	filename			Name of the file to delete
	 */
	static public function deleteFile($containerName, $filename) {

		//Verify all arguments were not empty
		if (!$containerName || !$filename) {return FALSE;}

		//Create the S3 handler object
		$awsFactory = getAWSFactory();
		$s3Handler = $awsFactory->get('s3');

		//Delete the file
		$s3Handler->deleteObject(array( 'Bucket'       => $containerName,
									    'Key'          => $filename));
	}


}


//---------------------------------------------------------------------------------------
//-------------------------- Helper Functions and Wrappers-------------------------------
//---------------------------------------------------------------------------------------

File: screenShotServer/sssEclipseProject/src/adshotrunner/utilities/FileStorageClient.java
package adshotrunner.utilities;

import java.io.File;

import com.amazonaws.services.s3.AmazonS3;
import com.amazonaws.services.s3.AmazonS3Client;
import com.amazonaws.services.s3.model.DeleteObjectRequest;
import com.amazonaws.services.s3.model.PutObjectRequest;

/**
 * 
 * The FileStorageClient saves and deletes files for system wide (and optionally public) access
 *
 */
public class FileStorageClient {
	
	//--------------------------------------------------------------------------------------
	//---------------------------------- Static Methods ------------------------------------
	//--------------------------------------------------------------------------------------
	//***************************** Public Static Methods **********************************
	/**
	 * Saves a file in the chosen container with the supplied filename.
	 * 
	 * @param containerName		Name of the container to save the file in
	 * @param fileToSave		Name and optional path of file to save
	 * @param newFilename		Name that the file to save should be stored under
	 */
	public static void saveFile(String containerName, String fileToSave, String newFilename) {
        
        //Instantiate an Amazon S3 client using AdShotRunner's AWS credentials
        AmazonS3 s3client = new AmazonS3Client(AWSConnector.getCredentials());
        
        //Upload the file
        File fileToUpload = new File(fileToSave);
        s3client.putObject(new PutObjectRequest(containerName, newFilename, fileToUpload));        
	}
	
	/**
	 * Deletes a file from the chosen container
	 * 
	 * @param containerName		Name of the container the file resides in
	 * @param filename			Name of the file to delete
	 */
	public static void deleteFile(String containerName, String filename) {
        
        //Instantiate an Amazon S3 client using AdShotRunner's AWS credentials
        AmazonS3 s3client = new AmazonS3Client(AWSConnector.getCredentials());
        
        //Delete the file
        s3client.deleteObject(new DeleteObjectRequest(containerName, filename));	        
	}
	
	
}
